hack/update-helm-index.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash -e
+
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o pipefail
+
+asset_urls="$@"
+
+git checkout gh-pages
+
+pushd stable > /dev/null
+# Download charts from release assets
+for asset_url in $asset_urls; do
+    if ! echo "$asset_url" | grep -q '.*tgz$'; then
+        echo "Skipping $asset_url, does not look like a Helm chart archive"
+        continue
+    fi
+
+    echo "Downloading $asset_url..."
+    curl -sSfLO $asset_url
+    # We rely on all release assets having the same baseurl
+    download_baseurl=`dirname $asset_url`
+done
+
+if [ -z "$download_baseurl" ]; then
+    echo "No Helm chart release assets found"
+    exit 0
+fi
+
+echo "Updating helm index"
+helm repo index stable --merge index.yaml --url https://nvidia.github.io/k8s-device-plugin/stable
```

Comment:
Is this what the `build-index.sh` script in `gh-pages` does?

Reply:
no, `build-index.sh` is

```bash
#!/usr/bin/env bash

helm repo index stable --url https://nvidia.github.io/k8s-device-plugin/stable

cp -f stable/index.yaml index.yaml
```

Reply:
`--merge string   merge the generated index into the given index`

Reply:
ideally, we could remove `build-index.sh` @ `gh-pages` once this PR is in.

Reply:
My question is shouldn't we keep the functionality but move it to main?